usaspending_api/awards/v2/filters/matview_filters.py

```diff
@@ -63,28 +63,33 @@ def matview_search_filter(filters, model):
         if key not in key_list:
             raise InvalidParameterException('Invalid filter: ' + key + ' does not exist.')
 
-        if key == "keyword":
-            keyword = value
-            upper_kw = keyword.upper()
-
-            # keyword_string & award_id_string are Postgres TS_vectors.
-            # keyword_string = recipient_name + naics_code + naics_description + psc_description + awards_description
-            # award_id_string = piid + fain + uri
-            compound_or = Q(keyword_ts_vector=keyword) | \
-                Q(award_ts_vector=keyword) | \
-                Q(recipient_unique_id=upper_kw) | \
-                Q(parent_recipient_unique_id=keyword)
-
-            if keyword.isnumeric():
-                compound_or |= Q(naics_code__contains=keyword)
-
-            if len(keyword) == 4 and PSC.objects.all().filter(code__iexact=keyword).exists():
-                compound_or |= Q(product_or_service_code__iexact=keyword)
-
-            queryset = queryset.filter(compound_or)
+        if key == "keywords":
+            def keyword_parse(keyword):
+                # keyword_string & award_id_string are Postgres TS_vectors.
+                # keyword_string = recipient_name + naics_code + naics_description
+                #     + psc_description + awards_description
+                # award_id_string = piid + fain + uri
+                filter_obj = Q(keyword_ts_vector=keyword) | \
+                    Q(award_ts_vector=keyword)
+                if keyword.isnumeric():
+                    filter_obj |= Q(naics_code__contains=keyword)
+                if len(keyword) == 4 and PSC.objects.all().filter(code__iexact=keyword).exists():
+                    filter_obj |= Q(product_or_service_code__iexact=keyword)
+
+                return filter_obj
+
+            filter_obj = Q()
+            for keyword in value:
+                filter_obj |= keyword_parse(keyword)
+            potential_DUNS = list(filter((lambda x: len(x) > 7 and len(x) < 10), value))
```

Comment:
I believe DUNS should always be a 9 digits. Could you provide an example when it's 8? I couldn't find one in staging. See subawards line 230-231:
```
if len(recipient_string) == 9:
    filter_obj |= Q(recipient__recipient_unique_id__iexact=recipient_string)
```